File: python/HelloWorld.py
from pyspark import SparkConf, SparkContext

# TODO: Exercise i-4:
# Launch standalone project
conf = SparkConf().setAppName("HelloWorld")
sc = SparkContext(conf = conf)
lines = sc.textFile("README.md")
pythonLines = lines.filter(lambda line: "Python" in line)
result = pythonLines.collect()
for line in result:
    print(line)

File: src/main/java/org/nchc/spark/java/exercise/HelloWorld.java
package org.nchc.spark.java.exercise;

import org.apache.log4j.Logger;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.List;

/**
 * Created by ogre0403 on 2016/1/23.
 */
public class HelloWorld {
    private static Logger logger = Logger.getLogger(HelloWorld.class);
    public static void main(String[] args) {
        String inputFile = args[0];
        SparkConf conf = new SparkConf().setAppName("HelloWorld").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> lines = sc.textFile(inputFile);

        JavaRDD<String> pythonLines = lines.filter(
            null
            //TODO: Exercise i-3, i-4
            // filter out line containing "Python"
            // delete null and replaced by appropriate Function class
        );

        List<String> result = pythonLines.collect();

        for(String s: result)
            logger.info(s);
    }
}
